### PR TITLE
updating observability export to external endpoints

### PR DESCRIPTION
**Pull request by dockerymick**
https://app.zenhub.com/workspaces/engineering-backlog-do-not-delete-604fab62d4b98d00150a2854/issues/stolostron/backlog/20624

### CODE REVIEW

**Comment by marcolan018** [comment]
@dockerymick 

The confgiuration for the endpoint is not part of MCO CR. It's a separate secret. So I think we should not put it in the observability API doc? Or add it as an appendix of the API doc?

And also we really need update in MCO API side, sorry I missed that part in my original post. Please check the content below:

In `storageConfig`, adds a new item `writeStorage `
Name | Description | Schema
 --- | --- | --- 
writeStorage(optional) | A list access info of write endpints | []WriteStorage

And the definition `WriteStorage` is as below:
Name | Description | Schema
 --- | --- | --- 
name(required)|The name of the secret with endpoint access info|string
key(required)|The key of the secret to select from|string

File: apis/observability.json.adoc [hunk 1]
```diff
@@ -109,6 +109,17 @@ __required__|Parameters describing the MultiClusterObservability resource to be
...
+    "http_client_config":
+      "basic_auth":
+        "username":
+        "password":
+      "tls_config":
+        "secret_name":
+        "ca_file_key":
+        "cert_file_key":
+        "key_file_key":
+        "insecure_skip_verify":
```
Comment (marcolan018) [inline apis/observability.json.adoc]: should not be added here

File: apis/observability.json.adoc [hunk 2]
```diff
@@ -284,8 +295,49 @@ __optional__|The advanced configuration settings for observability. |<<_rhacm-do
...
+_optioal_|Name of the secret that contains certificates.|string
+|**ca_file_key** +
+_optional_|Key of the CA certificate in the secret.|string
+|**cert_file_key** +
+_optional_|Key of the client certificate in the secret.|string
+|**key_file_key** +
+_optional_|Key of the client key in the secret.|string
+|**insecure_skip_verify** +
+_optional_|Parameter to skip the verification for target certificate.|bool
+|===
```
Comment (marcolan018) [inline apis/observability.json.adoc]: should not be added here

**Review by marcolan018 — APPROVED** [review]

**Comment by dockerymick** [comment]
@marcolan018 thanks for your explanation. In my opinion, the table for writeStorage is not necessary. The table is disruptive. I think it is best to keep the explanation in the docs. However, I did add the others tables

**Comment by dockerymick** [comment]
Made a couple of updates. Merging this PR

**Comment by marcolan018** [comment]
Hi @dockerymick 
the changes of writeStorage in observability API doc is necessary.  It’s a real change on the API, cannot be ignored. From user perspective, if they want to enable remote endpoint export, they need to firstly create the secret. For the content of the secret, you already added the description. Secondly, they also need to add the writeStorage part in MCO CR. So we need to add the related description about this. How to specify the secret name in MCO CR and the related key.
The similar case is the object storage configuration. In our sample, the secret thanos-object-stoage contains all necessary info about how to access S3 object storage. And in MCO CR, also need to specify the secret info as below, And this should be included in observability api doc

```
    metricObjectStorage:
      name: thanos-object-storage
      key: thanos.yaml
```

So the desired update of observability API doc is as below:

In `storageConfig`, adds a new item `writeStorage `
Name | Description | Schema
 --- | --- | --- 
writeStorage(optional) | A list access info of write endpints | []WriteStorage

And the definition `WriteStorage` is as below:
Name | Description | Schema
 --- | --- | --- 
name(required)|The name of the secret with endpoint access info|string
key(required)|The key of the secret to select from|string

**Pull request merged**